packages/compass-saved-aggregations-queries/src/components/edit-item-modal.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect } from 'react';
+import { ConfirmationModal, TextInput } from '@mongodb-js/compass-components';
+import { connect } from 'react-redux';
+import type { MapDispatchToProps, MapStateToProps } from 'react-redux';
+import type { RootState } from '../stores';
+import type { UpdateItemAttributes } from '../stores/edit-item';
+import type { Item } from '../stores/aggregations-queries-items';
+import { cancelEditItem, updateItem } from '../stores/edit-item';
+
+type EditItemModalProps = {
+  isModalOpen: boolean;
+  item?: Pick<Item, 'id' | 'name' | 'type'>;
+  onSubmit(id: string, attributes: UpdateItemAttributes): void;
+  onCancel: () => void;
+};
+
+const EditItemModal: React.FunctionComponent<EditItemModalProps> = ({
+  isModalOpen,
+  item,
+  onSubmit,
+  onCancel,
+}) => {
+  const [name, setName] = useState(item?.name ?? '');
+  useEffect(() => {
+    setName(item?.name ?? '');
+  }, [item]);
+
+  if (!item) {
+    return <></>;
+  }
+
+  const title = `Edit ${item.type === 'query' ? 'query' : 'aggregation'}`;
+  const onConfirm = () => {
+    return onSubmit(item.id, { name });
+  };
+  const isSubmitDisabled = () => {
+    return !name || name === item.name;
+  };
+  return (
+    <ConfirmationModal
+      open={isModalOpen}
+      onCancel={onCancel}
+      onConfirm={onConfirm}
+      data-testid="edit-item-modal"
+      title={title}
+      buttonText="Update"
+      submitDisabled={isSubmitDisabled()}
+    >
+      <form
+        onSubmit={(event) => {
+          event.preventDefault();
+          if (!isSubmitDisabled()) {
+            onConfirm();
+          }
+        }}
+      >
+        <TextInput
+          aria-label="Name"
+          label="Name"
+          name="name"
+          placeholder="Name"
+          onChange={(event) => {
+            setName(event.target.value);
+          }}
+          value={name}
+        />
+      </form>
+    </ConfirmationModal>
```

Comment:
Asked leafygreen about that yesterday and they suggest not to use confirmation modal for forms (see [this thread](https://mongodb.slack.com/archives/CFFA5BS79/p1644944835907879?thread_ts=1644942475.749649&cid=CFFA5BS79)) and instead build it ourselves from other leafygreen components